sdk/dapp-kit/src/hooks/wallet/useSignAndExecuteTransactionBlock.ts

```diff
@@ -39,43 +40,71 @@ type UseSignAndExecuteTransactionBlockMutationOptions = Omit<
 		unknown
 	>,
 	'mutationFn'
->;
+> & {
+	executeFromWallet?: boolean;
+};
 
 /**
  * Mutation hook for prompting the user to sign and execute a transaction block.
  */
 export function useSignAndExecuteTransactionBlock({
 	mutationKey,
+	executeFromWallet,
 	...mutationOptions
 }: UseSignAndExecuteTransactionBlockMutationOptions = {}) {
 	const currentWallet = useCurrentWallet();
 	const currentAccount = useCurrentAccount();
+	const client = useSuiClient();
 
 	return useMutation({
 		mutationKey: walletMutationKeys.signAndExecuteTransactionBlock(mutationKey),
-		mutationFn: async (signAndExecuteTransactionBlockArgs) => {
+		mutationFn: async ({ requestType, options, ...signTransactionBlockArgs }) => {
 			if (!currentWallet) {
 				throw new WalletNotConnectedError('No wallet is connected.');
 			}
 
-			const signerAccount = signAndExecuteTransactionBlockArgs.account ?? currentAccount;
+			const signerAccount = signTransactionBlockArgs.account ?? currentAccount;
 			if (!signerAccount) {
 				throw new WalletNoAccountSelectedError(
 					'No wallet account is selected to sign and execute the transaction block with.',
 				);
 			}
 
-			const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock'];
+			if (executeFromWallet) {
+				const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock'];
+				if (!walletFeature) {
+					throw new WalletFeatureNotSupportedError(
+						"This wallet doesn't support the `signAndExecuteTransactionBlock` feature.",
+					);
+				}
+
+				return walletFeature.signAndExecuteTransactionBlock({
+					...signTransactionBlockArgs,
+					account: signerAccount,
+					chain: signTransactionBlockArgs.chain ?? signerAccount.chains[0],
+					requestType,
+					options,
+				});
+			}
+
+			const walletFeature = currentWallet.features['sui:signTransactionBlock'];
 			if (!walletFeature) {
 				throw new WalletFeatureNotSupportedError(
 					"This wallet doesn't support the `signAndExecuteTransactionBlock` feature.",
 				);
 			}
 
-			return await walletFeature.signAndExecuteTransactionBlock({
-				...signAndExecuteTransactionBlockArgs,
+			const { signature, transactionBlockBytes } = await walletFeature.signTransactionBlock({
+				...signTransactionBlockArgs,
 				account: signerAccount,
-				chain: signAndExecuteTransactionBlockArgs.chain ?? signerAccount.chains[0],
+				chain: signTransactionBlockArgs.chain ?? signerAccount.chains[0],
+			});
+
+			return client.executeTransactionBlock({
```

Comment:
what's the use case for signing with the wallet and then executing from the dApp? when would we want to sign and execute (i.e., client.signAndExecuteTransaction) all from the dApp?

just trying to make sure that this branched logic belongs in this hook since these are wallet hooks after all

Reply:
this was something @Jordan-Mysten and I were just talking about.

Basically, you will see your transaction more quickly if you are executing against the same RPC node as you are fetching from.  Generally this means you want to execute agains the node configured in your dApp rather than the one in your wallet.

@Jordan-Mysten was asking me to update the examples to always use `useSignTransactionBlock` and then manually execute as a separate step, because this is almost always what you want.   Handling this automatically in the hook will be cleaner for dApp builders.  I can't think of reason you'd want to execute from the wallet rather than the dApp, but it's something we want to collect some feedback on to be sure.

Reply:
I think Hayes explained this well, but in general I wouldn't think of these as "wallet hooks" I would instead think of these as "dapp hooks". Often in dapps you'll sign and then execute a transaction, so it probably makes sense to expose that as a hook. Just because it doesn't map 1:1 with wallet features doesn't mean we shouldn't offer it.